### ISSUE TITLE
Replace the web server framework

### ISSUE THREAD

**Issue by yaleman** [kanidm/kanidm#1787]
**Is your feature request related to a problem? Please describe.**

`tide` is unmaintained, time to find a new HTTPS service.

**Describe the solution you'd like**

Migrate from `tide` to `axum`

**Describe alternatives you've considered**

- Rocket ("easy, but can be slow")
- Actix (seems good, has some curious ways, happy to look at it if Axum doesn't work)
- writing something entirely from scratch (NO, WILLIAM!)

**Additional context**

- https://www.arewewebyet.org
- https://blog.logrocket.com/top-rust-web-frameworks/

** Steps **

- [ ] Ensure *all* the current routes are tested in some way
- [ ] replace tide
- [ ] ???
- [ ] 🦀 🍰 🥳 
- [ ] ensure HTTP/2 works properly.
- [ ] ensure all endpoints work the same.

**Comment by Firstyear** [comment]
Gonna put it out here now, axum is the choice I'm going with.

* Actix - no go, we used it in the past and I had bad experiences.
* Rocket - I'm not a fan of it's architecture, and slow is an issue for us in future.
* Axum - Huge support based, based on hyper which is also part of reqwest, very popular, well maintained. Seems to have the wind blowing there. 

I already wrote the support for axum to support openssl tbh :) so I'm already making some moves here. I'm rewriting some of my personal apps to axum to get experience and resolve issues. 

So I think this is going to happen sooner than later.

Re route testing a lot of those are covered in the proto tests already, but not all. Could be a good chance to expand that.

**Comment by yaleman** [comment]
Yep, Axum's something I've had experience with and I'm expanding the testing to at least *hit* the routes to ensure they exist and kinda make sense.

Code-coverage, zomg.
